kolibri/core/tasks/api.py

```diff
@@ -1101,6 +1111,76 @@ def startdeletefacility(self, request):
         resp = _job_to_response(facility_queue.fetch_job(job_id))
         return Response(resp)
 
+    @decorators.action(
+        methods=["post"],
+        detail=False,
+        permission_classes=[
+            IsSuperuser | NotProvisionedCanPost | LODUserHasSyncPermissions
+        ],
+    )
+    def startprovisionsoud(self, request):
+        baseurl = request.data.get("baseurl", None)
+        facility_id = request.data.get("facility_id", None)
+        username = request.data.get("username", None)
+        password = request.data.get("password", None)
+        user_id = request.data.get("user_id", None)
+        device_name = request.data.get("device_name", None)
+        if user_id is None:
+            auth_url = urljoin(baseurl, reverse("kolibri:core:session-list"))
```

Comment:
I know I suggested this, but wonder if we should create a separate public endpoint for this for future stability?

I had suggested something similar here: https://github.com/learningequality/kolibri/issues/8170 using Basic Auth to allow return of either one user or multiple users, depending on permissions.

Reply:
Done in https://github.com/learningequality/kolibri/pull/8185/commits/47b486067fc25bf0748c326f9f07e1dd4e0c9b0a

Reply:
Adapted BasicAuth to include the facility_id as a realm when login https://github.com/learningequality/kolibri/blob/a56d04850e9c765b50a94ec0b25c0279ed197dfb/kolibri/core/auth/api.py#L82-L106